cmd/vic-machine/converter/converter.go

```diff
@@ -0,0 +1,508 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package converter
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"reflect"
+	"strconv"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/vic/lib/install/data"
+	"github.com/vmware/vic/pkg/ip"
+	"github.com/vmware/vic/pkg/trace"
+)
+
+const (
+	cmdTag         = "cmd"
+	labelTag       = "label"
+	parentTagValue = "parent"
+
+	optionSeparator = "-"
+
+	keyAfterValueLabel = "value-key"
+	valueAfterKeyLabel = "key-value"
+)
+
+var (
+	kindConverters = make(map[reflect.Kind]converter)
+	typeConverters = make(map[string]converter)
+	labelHandlers  = make(map[string]labelConverter)
+
+	ConverterLogLevel = log.InfoLevel
+)
+
+type converter func(src reflect.Value, prefix string, tags reflect.StructTag, dest map[string][]string) error
+type labelConverter func(dest map[string][]string, key string) error
+
+func init() {
+	kindConverters[reflect.Struct] = convertStruct
+	kindConverters[reflect.Slice] = convertSlice
+	kindConverters[reflect.Map] = convertMap
+	kindConverters[reflect.String] = convertString
+	kindConverters[reflect.Ptr] = convertPtr
+	kindConverters[reflect.Int] = convertPrimitive
+	kindConverters[reflect.Int8] = convertPrimitive
+	kindConverters[reflect.Int16] = convertPrimitive
+	kindConverters[reflect.Int32] = convertPrimitive
+	kindConverters[reflect.Int64] = convertPrimitive
+	kindConverters[reflect.Bool] = convertPrimitive
+	kindConverters[reflect.Float32] = convertPrimitive
+	kindConverters[reflect.Float64] = convertPrimitive
+
+	typeConverters["url.URL"] = convertURL
+	typeConverters["net.IPNet"] = convertIPNet
+	typeConverters["net.IP"] = convertIP
+	typeConverters["ip.Range"] = convertIPRange
+	typeConverters["data.NetworkConfig"] = convertNetwork
+	typeConverters["data.ContainerNetworks"] = convertContainerNetworks
+
+	labelHandlers[keyAfterValueLabel] = keyAfterValueLabelHandler
+	labelHandlers[valueAfterKeyLabel] = valueAfterKeyLabelHandler
+}
+
+// DataToOption convert data.Data structure to vic-machine create command options based on tags defined in data.Data structure
+// Note: need to make sure the tags are consistent with command line option name
+func DataToOption(data *data.Data) (map[string][]string, error) {
+	defer log.SetLevel(log.GetLevel())
+	log.SetLevel(ConverterLogLevel)
```

Comment:
This will set the global log level, not just here. If you want to use different log levels for this package, then use package level logger, no the global one.